src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -107,12 +109,22 @@ public function __construct(array $options = [])
         }
 
         if (!($options['disable_dotenv'] ?? false) && isset($options['project_dir']) && !class_exists(MissingDotenv::class, false)) {
-            (new Dotenv($envKey, $debugKey))
+            $overrideExistingVars = $options['dotenv_overload'] ?? false;
+            $dotenv = (new Dotenv($envKey, $debugKey))
                 ->setProdEnvs((array) ($options['prod_envs'] ?? ['prod']))
-                ->usePutenv($options['use_putenv'] ?? false)
-                ->bootEnv($options['project_dir'].'/'.($options['dotenv_path'] ?? '.env'), 'dev', (array) ($options['test_envs'] ?? ['test']), $options['dotenv_overload'] ?? false);
+                ->usePutenv($options['use_putenv'] ?? false);
 
-            if (isset($this->input) && ($options['dotenv_overload'] ?? false)) {
+            $dotenv->bootEnv($options['project_dir'].'/'.($options['dotenv_path'] ?? '.env'), 'dev', (array) ($options['test_envs'] ?? ['test']), $overrideExistingVars);
+
+            if (\is_array($options['dotenv_extra_paths'] ?? null) && [] !== $options['dotenv_extra_paths']) {
```

Comment:
we don't do empty-values comparisons since they are just visual noise ;)
```suggestion
            if (\is_array($options['dotenv_extra_paths'] ?? null) && $options['dotenv_extra_paths']) {
```

Reply:
thank you @nicolas-grekas for reviewing this, I've implemented the change